[WIP] Update 0.3.10

ChrisAlderson opened the pull request.

xaiki committed:
start gulp integration stolen from CE fork

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

krikmo committed:
Improve grammar for contributions guide

xaiki committed:
Merge pull request #291 from krikmo/master

Improve grammar for contributions guide

vankasteelj committed:
jshint src/app/app.js (credit @alxhotel)

vankasteelj committed:
jshint src/app/bootstrap.js (credits @alxhotel)

vankasteelj committed:
jhint src/app/httpapi.js (credits @alxhotel)

vankasteelj committed:
jshint src/app/lib/device/ext-player.js (credits @alxhotel)

vankasteelj committed:
jshint src/app/lib/device/generic.js (credits @alxhotel)

vankasteelj committed:
split getDataFromProvider() (credits @alxhotel)

vankasteelj committed:
jshint src/app/lib/providers/generic.js (credits @alxhotel)

vankasteelj committed:
jshint src/app/lib/providers/torrent_cache.js (credits @alxhotel)

vankasteelj committed:
jshint src/app/lib/providers/trakttv.js (credits @alxhotel)

vankasteelj committed:
jshint src/app/lib/providers/watchlist.js (credits @alxhotel)

vankasteelj committed:
jshint src/app/lib/streamer.js (credits @alxhotel)

vankasteelj committed:
jshint src/app/updater.js (credits @alxhotel)

vankasteelj committed:
jshint src/app/settings.js (credits @alxhotel)

vankasteelj committed:
setTimeout on the request object (credits @alxhotel)

vankasteelj committed:
fix jshint in streamer :p

vankasteelj committed:
update devDependencies

vankasteelj committed:
'safe' update of dependencies

- nedb needs testing before updating
- we now force security update on moment
- gitlab can't be tested atm as we use github

MiguelSR committed:
Fixed Issue #17: cache directory couldn't be change because overriden Settings variable

xaiki committed:
don't hardcode ysubs

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

team-pct committed:
Fix magnet link

Fix Magnet Link

vankasteelj committed:
simplify AddGhostsToBottomRow

+ bonus: be prepared in case of hidden items

vankasteelj committed:
(forgot to update NUM_MOVIES_IN_ROW, oups)

Commit:
Add time-grunt to measure Grunt execution time

xaiki committed:
Merge pull request #302 from AdvancedCori/master

Add time-grunt to measure Grunt execution time

z11h committed:
use minified versions of js files

This change *should* help load times.

vankasteelj committed:
new watchlist (#301)

vankasteelj committed:
watchlist: new trakt restrictions

vankasteelj committed:
update watchlist using trakt.tv module

z11h committed:
add prestart script to package.json

z11h committed:
update README.md

vankasteelj committed:
jshint

vankasteelj committed:
memoize watchlist

z11h committed:
add the add -g grunt-cli bower

shouldn't have been removed in the first place :smile:

z11h committed:
Merge pull request #7 from butterproject/master

pull in JS Hint fix (for Travis)

vankasteelj committed:
fix cache not being invalidated

xaiki committed:
export package.json as a global App.Npm object

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

z11h committed:
add grunt build and start tasks to prestart script

z11h committed:
remove unneeded quickstart tasks

this commit removes some unneeded tasks that automatically get run by the npm prestart script :smile:

z11h committed:
update README.md

xaiki committed:
remove old cruft

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
this one is ok

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
Merge pull request #314 from ZakariaRidouh/patch-4

use pre start script to install packages

xaiki committed:
good faith commit, remove search using kat api, we don't use it anyway

We got a DMCA takedown notice, apparently using kat to search for metadata is not OK, removing it for now to show good faith

xaiki committed:
Merge pull request #307 from ZakariaRidouh/patch-2

use minified versions of js files

vankasteelj committed:
update Common.matchTorrent

vankasteelj committed:
use /right/ minified versions of js files

vankasteelj committed:
use correct exec:linux32/64 paths

This is dev env, nothing to do with builds/

vankasteelj committed:
update watchlist.js + remove leftover erroring

VitorVRS committed:
Fixed image selection for bookmarked movies

vankasteelj committed:
use file:// protocol on local video files (fix GET_ERR app://host)

xaiki committed:
Merge pull request #332 from butterproject/fix/linux_dragdrop

use file:// protocol on local video files (fix GET_ERR app://host)

vankasteelj committed:
Use newer upnp-mediarenderer-client (dlna cast)

xaiki committed:
Merge pull request #333 from butterproject/fix/dlna

Use newer upnp-mediarenderer-client (dlna cast)

greenkeeperio-bot committed:
chore(package): update torrent-tracker-health to version 1.0.0

http://greenkeeper.io/

vankasteelj committed:
modify for torrent-tracker-health 1.0.0

vankasteelj committed:
why stick to 1.0.0?

vankasteelj committed:
Merge pull request #338 from butterproject/greenkeeper-torrent-tracker-health-1.0.0

Update torrent-tracker-health to version 1.0.0 🚀

vankasteelj committed:
fix bower_clean due to using minified js

xaiki committed:
guard against undefined providers

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

vankasteelj committed:
fix grunt build (closes #341)

vankasteelj committed:
basic gulpfile

vankasteelj committed:
Merge branch 'master' into gulp

vankasteelj committed:
more work on gulp

vankasteelj committed:
npm scripts

vankasteelj committed:
injectgit task

vankasteelj committed:
gulp dist (+makensis/compress)

vankasteelj committed:
compress using lzma on unix

greenkeeperio-bot committed:
chore(package): update bower to version 1.7.9

http://greenkeeper.io/

vankasteelj committed:
gulpfile logging & queuing

vankasteelj committed:
better gulp logging

vankasteelj committed:
deb task

vankasteelj committed:
all working so far

vankasteelj committed:
gulp jshint & pre-commit hook

vankasteelj committed:
set up gulp beautifier task

vankasteelj committed:
testing pre-commit hint&beautify

vankasteelj committed:
gulp pre-commit correcte does jshint & jsbeautifier

vankasteelj committed:
default is help

vankasteelj committed:
reorganize tasks & use 'build' in 'dist'

vankasteelj committed:
gulp clean:build

vankasteelj committed:
gulp-load-plugins ftw

vankasteelj committed:
should fix guppy & pre-commit hook

vankasteelj committed:
format help with [].join()

vankasteelj committed:
better build without prune

xaiki committed:
Merge pull request #353 from butterproject/gulp

Gulp

vankasteelj committed:
spawn is somewhat dump

xaiki committed:
code readability: handle failure first

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
make returns consistent

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
jshint

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
arrow functions for all

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
make jshint happy with es6

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
make clean code cleaner

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
beautify

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

vankasteelj committed:
Merge pull request #357 from butterproject/xaiki/gulp-npm

some gulp fixes, mostly style.

vankasteelj committed:
Merge pull request #358 from butterproject/xaiki/gulp-es6

Xaiki/gulp es6

vankasteelj committed:
use prebuilt nwjs, enhance parsePlatforms()

vankasteelj committed:
es6 down the road (no import though)

- also rewrite parseReqDeps
- correctly log failing bash scripts

vankasteelj committed:
regroup del wrapper+logger, comment customs

vankasteelj committed:
comment gulp further

vankasteelj committed:
fix deb name

vankasteelj committed:
travis gulp

vankasteelj committed:
travis npm@3, update devdeps (yargs)

xaiki committed:
implement inject-git with git-rev

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
beautify

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

vankasteelj committed:
write to .git.json

vankasteelj committed:
Merge pull request #363 from butterproject/xaiki/gulp-git

Xaiki/gulp git

vankasteelj committed:
Revert "Xaiki/gulp git"

vankasteelj committed:
Merge pull request #364 from butterproject/revert-363-xaiki/gulp-git

Revert "Xaiki/gulp git"

vankasteelj committed:
injectgit task

vankasteelj committed:
rearrange a few things in promiseCallback

vankasteelj committed:
kill CI mails [ci-skip]

vankasteelj committed:
remove beautify hooks

vankasteelj committed:
remove guppy-git

vankasteelj committed:
Merge pull request #366 from butterproject/vkj/gulp

remove beautify hooks

vankasteelj committed:
Merge branch 'greenkeeper-bower-1.7.9'

VitorVRS committed:
Fixed TVST not restoring token (#113)

Moved logic from TVShowTime constructor to another method, which now can be calld from its constructor and Database (after loading settings)
Add new method call on database file to restore token from settings.

vankasteelj committed:
nw-builder log errors correctly

vankasteelj committed:
fix #375

replace nw-builder with a fork (PR waiting merge) to allow the use of a
custom 'currentPlatform' flag, allowing to run 32bits executable on 64bits
machines

todo: switch back to official if/when pr is merged

vankasteelj committed:
gulp run fixup

use non-hardcoded spawn replacement for nw.run()
* conveniently allows restarting the app
* full error log
* report node logs to main console
* fixes #381
* fixes #375

vankasteelj committed:
es2015ify run task

vankasteelj committed:
add missing resolve()

was causing compress task to hang and fail

xaiki committed:
Merge pull request #382 from butterproject/nw-builder

fix #375

vankasteelj committed:
fix wcjs mess (again)

vankasteelj committed:
Show app in 'select application' popup (linux)

Adding %U as a param to Exec allows app to show up in Select Application popup.
Tested on Fedora 23 with Gnome and Ubuntu 16.04 unity.

credits for idea @petre2dor

vankasteelj committed:
Revert "fix #375"

vankasteelj committed:
Merge pull request #390 from butterproject/revert-382-nw-builder

Revert "fix #375"

vankasteelj committed:
Merge pull request #383 from butterproject/gulp/run

gulp run fixup

vankasteelj committed:
Merge pull request #389 from butterproject/forks/merge-back

Forks/merge back

vankasteelj committed:
fix title being replaced on subs not found

vankasteelj committed:
display local file as fully downloaded

vankasteelj committed:
html on metadatacheck (for &amp; and similar)

vankasteelj committed:
use harmony flag

some modules changed to es6, fixes #391

greenkeeperio-bot committed:
chore(package): update peerflix to version 0.35.0

https://greenkeeper.io/

greenkeeperio-bot committed:
chore(package): update opensubtitles-api to version 3.1.0

https://greenkeeper.io/

vankasteelj committed:
modifications for opensubtitles-api 3.x

- login() returns object instead of string

vankasteelj committed:
Merge branch 'greenkeeper-peerflix-0.35.0'

vankasteelj committed:
remove node-tvdb (unused)

vankasteelj committed:
remove xmlbuilder (unused)

team-pct committed:
Butter Updates

team-pct committed:
Butter Updates #2

xaiki committed:
introduce new lang-dropdown template

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
add nativeName() method to App.Localization

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
add lang_dropdown view

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
add lang model

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
make movie_details use the dropdown view

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

team-pct committed:
Butter Updates #3

xaiki committed:
return lang if no nativeLang is found

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
jshint

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
remove unused values

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

team-pct committed:
Audio Dropdown #2

team-pct committed:
Lang Improvement

VitorVRS committed:
Added gulp to devDependencies

team-pct committed:
Merge pull request #194 from VitorVRS/dev2

Added gulp to devDependencies

VitorVRS committed:
Fixed movie thumbnails error when using https

team-pct committed:
Merge pull request #203 from VitorVRS/dev2

Fixed movie thumbnails error when using https

Commit:
Reverted audio-dropdown and added anime provider

Commit:
resolved merge conflicts

team-pct committed:
Update installer_makensis.nsi

ChrisAlderson committed:
Updated package.json

Update bower dependencies

Commit:
fix windows installer

ChrisAlderson committed:
fix for torrent collection

ChrisAlderson committed:
fix for movie backdrop

ChrisAlderson committed:
fix for movie sub dropdown not clickable

Commit:
more fixes for torrent collection

Commit:
removed local ysubs provider

team-pct committed:
Update opensubtitles.js

team-pct committed:
OSX64

ChrisAlderson committed:
update nwjs to version 0.18.1

team-pct committed:
Frame is deprecated

ChrisAlderson committed:
Revert "Frame is deprecated"

This reverts commit 7c8c635b79f1d3551ebd10c83eb394d26180ed7d.

ChrisAlderson committed:
removed deprecated toolbar

ChrisAlderson committed:
removed deprecated unmaximize

ChrisAlderson committed:
fixed importing and exporting the settings. credits should go to @vzamanillo

ChrisAlderson committed:
fixed importing and exporting settings, credits go to @vzamanillo.

vzamanillo committed:
Nw18

ChrisAlderson committed:
updated OS UA

VitorVRS commented on src/app/lib/providers/trakttv.js:
```diff
@@ -515,6 +516,25 @@
+                AdvSettings.set('traktTokenTTL', '');
+                AdvSettings.set('traktTokenRefresh', '');
+                defer.reject(err);
+>>>>>>> a383973... nw.js => 0.13 migration due to architecture Changes
```
Merge conflicts here.

team-pct committed:
Revert "Nw18"

This reverts commit 3217d8cf1e6a7b5c3e6ff5333152ba33fb8fc642.

vzamanillo committed:
Improved switchRating code

Improved switchRating code (sorted) for movies and TV shows

vzamanillo committed:
jshint W116 fix

W116: Expected '!==' and instead saw '!='.

ChrisAlderson committed:
Revert "Revert "Nw18""
yea a reversion of a reversion.
This reverts commit 34cad517998e336425b18d610603e560fb8374cd.

ChrisAlderson committed:
resolved merge conflicts

ChrisAlderson committed:
replaced gui with nw

ChrisAlderson committed:
fixed connecting to tvshowtime

ChrisAlderson committed:
fixed connecting to trakt.tv

ChrisAlderson committed:
replaced gui with nw

ChrisAlderson committed:
indents

ChrisAlderson committed:
removed focus function

team-pct committed:
Fix sub Issues on new Builds

team-pct committed:
Merge pull request #254 from popcorn-official/subfix

Fix sub Issues on new Builds

team-pct committed:
Adding win64

ChrisAlderson committed:
removed ysubs (down)

ChrisAlderson committed:
removed sub selection for movies

ChrisAlderson committed:
movie using default subtitle language

team-pct committed:
Update packages

team-pct committed:
Merge pull request #259 from popcorn-official/update/packages

Update packages

vankasteelj committed:
Fix Sub DL Credits @vankasteelj

team-pct committed:
Update externals des

vzamanillo committed:
Default background on loading movie or show

If no background image or image could not be loaded we set the default background image on loading view.

wallydz committed:
Fix Sub DL Credits @vankasteelj #2

wallydz committed:
Temp Fix for Sub on Movies Section
